File: models/Product.js
const mongoose = require('mongoose');

const productDetailsSchema = new mongoose.Schema({
    price: {
        type: String,
        required: true,
    },
    description: {
        type: String,
        required: true,
    },
});

const productSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
    },
    details: {
        type: productDetailsSchema,
        required: true,
    },
    stock: {
        type: Number,
        required: true,
        default: 0,
    },
    createdAt: {
        type: Date,
        default: Date.now,
    },
});

const Product = mongoose.model('Product', productSchema);

module.exports = Product;

File: Model/Product.cs
using Google.Cloud.Firestore;

namespace Product_Api.Model
{
    [FirestoreData]
    public class Product
    {
        [FirestoreDocumentId]
        public string Id { get; set; }

        [FirestoreProperty]
        public string Name { get; set; }

        [FirestoreProperty]
        public ProductDetails Details { get; set; }

        [FirestoreProperty]
        public int Stock { get; set; }

        [FirestoreProperty]
        public DateTime CreatedAt { get; set; }
    }
}